### PR TITLE
hooks to process fixtures - closes #121

### LINKED ISSUE #121: verify process existence in the beginning of adequate connection fixture
The problem is that one of tests in a sequence of them may cause e.g. `redis_proc` to stop. The next test using the same fixture will fail while trying to connect to this redis. So it happens that the problem from one test leaks to the other and that is hard to debug. Sometimes even one of tests passes but in the end it makes redis/mongo/etc. to exit so the error will occur in the next test trying to connect to one of databases.

I see two solutions:
- connection fixtures (redisdb, rabbitmq, etc.) try once to restart the process if the connection fails. If second connection attempt ends with failure fixture should raise exception.
- process fixtures (redis_proc, rabbitmq_proc, etc.) should add a [hook](https://pytest.org/latest/writing_plugins.html#writing-hook-functions) (code ref: [hookspec.py](https://github.com/pytest-dev/pytest/blob/2.8.2/_pytest/hookspec.py)) that will be fired in the beginning or the end of test verifying if process is running and trying to restart it in case it was down. An advantage of the process fixtures hooks approach would be that whoever will use own connection fixture (and not the one in pytest-dbfixtues) he won't need to solve the issue on his own.

### PR DESCRIPTION
**Pull request by szwagier90**
Added hooks to all process fixtures to check if process is running, and
if not, try to start it again.

### CODE REVIEW

File: src/pytest_dbfixtures/factories/elasticsearch.py [hunk 1]
```diff
@@ -109,6 +109,11 @@ def finalize_elasticsearch():
             shutil.rmtree(home_path)
 
         request.addfinalizer(finalize_elasticsearch)
+
+        def pytest_runtest_setup(item):
+            if elasticsearch_executor.running() == False:
```
Comment (not-raspberry) [inline src/pytest_dbfixtures/factories/elasticsearch.py]: You are probably missing Pylama hook in the project.

`$ pylama --hook`

Then it won't let you commit code that doesn't lint.

File: src/pytest_dbfixtures/factories/postgresql.py [hunk 2]
```diff
@@ -203,6 +203,10 @@ def stop_server_and_remove_directory():
         if '-w' in config.postgresql.startparams:
             wait_for_postgres(logfile_path, START_INFO)
 
+        def pytest_runtest_setup(item):
```
Comment (not-raspberry) [inline src/pytest_dbfixtures/factories/postgresql.py]: Is there a way not to repeat the code for each fixture?

**Comment by not-raspberry** [comment]
We'll also need some tests for that. How else do we know the solution works?

File: CHANGES.rst [hunk 3]
```diff
@@ -1,6 +1,11 @@
 CHANGES
 =======
 
+0.13.0
```
Comment (fizyk) [inline CHANGES.rst]: please do not assume next version, use unreleased

**Pull request closed without merging**